lib/app/features/wallet/views/pages/manage_coins/providers/manage_coins_provider.dart

```diff
@@ -1,56 +1,71 @@
 // SPDX-License-Identifier: ice License 1.0
 
+import 'package:hooks_riverpod/hooks_riverpod.dart';
 import 'package:ion/app/features/wallet/views/pages/manage_coins/model/manage_coin_data.dart';
 import 'package:ion/app/features/wallet/views/pages/manage_coins/providers/mock_data/manage_coins_mock_data.dart';
 import 'package:riverpod_annotation/riverpod_annotation.dart';
 
 part 'manage_coins_provider.g.dart';
 
-@riverpod
+@Riverpod(keepAlive: true)
 class ManageCoinsNotifier extends _$ManageCoinsNotifier {
   @override
   AsyncValue<List<ManageCoinData>> build() {
     return AsyncData<List<ManageCoinData>>(
-      List<ManageCoinData>.unmodifiable(<ManageCoinData>[]),
+      List<ManageCoinData>.unmodifiable(mockedManageCoinsDataArray),
     );
   }
 
-  Future<void> fetch({
-    required String searchValue,
-  }) async {
-    state = const AsyncLoading<List<ManageCoinData>>().copyWithPrevious(state);
-
-    // to emulate loading state
-    await Future<void>.delayed(const Duration(seconds: 1));
-
-    final lSearchValue = searchValue.trim().toLowerCase();
-
+  void switchCoin({required String coinId}) {
+    final currentList = state.value ?? <ManageCoinData>[];
     state = AsyncData<List<ManageCoinData>>(
       List<ManageCoinData>.unmodifiable(
-        mockedManageCoinsDataArray.where(
-          (ManageCoinData data) => data.coinData.name.toLowerCase().contains(lSearchValue),
-        ),
+        currentList.map((coin) {
+          if (coin.coinData.abbreviation == coinId) {
+            return coin.copyWith(isSelected: !coin.isSelected);
```

Comment:
Could we switch to a Map here so we don't have to map over entire list each time a single coin is toggled?